staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go

```diff
@@ -599,6 +605,22 @@ func (s preparedGenericAPIServer) Run(stopCh <-chan struct{}) error {
 		s.HandlerChainWaitGroup.Wait()
 	}()
 
+	if s.ShutdownLongRunningRequestsTimeout != time.Duration(0) {
+		go func() {
+			defer klog.V(1).InfoS("[graceful-termination] shutdown event - draining timed out", "name", drainedCh.Name())
+			defer drainedCh.Signal()
```

Comment:
I don't understand this. Let's say I set a super short "long-running-shutdown-period". Than we will quickly signal `drainedCh` here and will shutdown the apiserver without properly draining regular (non-long-running) requests.

That seems wrong to me.

In general, as long as I see a value for introducing timeout for drainign short-running requests, that's to significant extent a separate thing to me. This whole code is subtle enough, that I definitely don't want to do two separate things in the same PR.

I strongly believe that we should make this as simple to reason about as possible.
So let's instead:
- introduce a new signal for draining long-running requests
- use that signal here
- wait for that new signal in the above (before calling `drainedCh.Signal()` [to effectively avoid changing the semantic]

We can introduce a timeout on the draining, but let's not mix that into this PR.